tensorflow_lite_support/cc/test/task/processor/image_preprocessor_test.cc

```diff
@@ -0,0 +1,147 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/cc/task/processor/image_preprocessor.h"
+
+#include <fstream>
+#include <memory>
+
+#include "absl/flags/flag.h"
+#include "absl/status/status.h"
+#include "absl/strings/cord.h"
+#include "tensorflow/lite/core/shims/cc/shims_test_util.h"
+#include "tensorflow/lite/kernels/builtin_op_kernels.h"
+#include "tensorflow/lite/mutable_op_resolver.h"
+#include "tensorflow_lite_support/cc/port/gmock.h"
+#include "tensorflow_lite_support/cc/port/gtest.h"
+#include "tensorflow_lite_support/cc/port/status_matchers.h"
+#include "tensorflow_lite_support/cc/task/core/task_api_factory.h"
+#include "tensorflow_lite_support/cc/task/core/task_utils.h"
+#include "tensorflow_lite_support/cc/task/core/tflite_engine.h"
+#include "tensorflow_lite_support/cc/task/vision/utils/frame_buffer_common_utils.h"
+#include "tensorflow_lite_support/cc/test/test_utils.h"
+#include "tensorflow_lite_support/examples/task/vision/desktop/utils/image_utils.h"
+
+namespace tflite {
+namespace task {
+namespace processor {
+namespace {
+
+using ::testing::ElementsAreArray;
+using ::testing::HasSubstr;
+using ::testing::Optional;
+using ::tflite::support::kTfLiteSupportPayload;
+using ::tflite::support::StatusOr;
+using ::tflite::support::TfLiteSupportStatus;
+using ::tflite::task::JoinPath;
+using ::tflite::task::core::PopulateTensor;
+using ::tflite::task::core::TaskAPIFactory;
+using ::tflite::task::core::TfLiteEngine;
+using ::tflite::task::vision::DecodeImageFromFile;
+using ::tflite::task::vision::FrameBuffer;
+using ::tflite::task::vision::ImageData;
+
+constexpr char kTestDataDirectory[] =
+    "tensorflow_lite_support/cc/test/testdata/task/vision/";
+
+constexpr char kDilatedConvolutionModelWithMetaData[] = "dilated_conv.tflite";
+
+StatusOr<ImageData> LoadImage(std::string image_name) {
+  return DecodeImageFromFile(
+      JoinPath("./" /*test src dir*/, kTestDataDirectory, image_name));
+}
+
+class DynamicInputTest : public tflite_shims::testing::Test {
+ public:
+  void SetUp() {
+    engine_ = absl::make_unique<TfLiteEngine>();
+    engine_->BuildModelFromFile(JoinPath("./", kTestDataDirectory,
+                                         kDilatedConvolutionModelWithMetaData));
+    engine_->InitInterpreter();
+
+    SUPPORT_ASSERT_OK_AND_ASSIGN(preprocessor_,
+                                 ImagePreprocessor::Create(engine_.get(), {0}));
+  }
+
+ protected:
+  std::unique_ptr<ImagePreprocessor> preprocessor_ = nullptr;
+  std::unique_ptr<TfLiteEngine> engine_ = nullptr;
+};
+
+// See if input tensor dims signature for height and width is -1
+// because it is so in the model.
+TEST_F(DynamicInputTest, InputHeightAndWidthMutable) {
```

Comment:
I think I've commented this before. No need to test the input tensor dims for height / width of the test model. This test can be removed.